### ISSUE TITLE
Future-proof conversion via repr(C)-struct-tuple

### ISSUE THREAD

**Issue by lukaslueg** [lukaslueg/restruct#3]
We currently go from a `[u8; _]` directly to a normal `(...)`-tuple and back. This works as long as their sizes match exactly (transmute will compile-error otherwise) but still leaves room open for error: A future compiler might decide to re-order the memory representation of a tuple-type (e.g. in order to avoid excessive alignment). The result will be safe behavior but completely garbage outcome.

The `pack()`- and `unpack()`-function should indirect via a `repr(C) struct` when going from/to a `[u8;_]` and convert that value again. This should always work and, in case the types actually match up exactly, compile to nothing.

**Comment by bernii** [comment]
Seems that nightly compiler is doing that - the order changes when going from tuple to `[u8;_]`
tested on `1.68.0-nightly (c994a4a63 2022-12-18)`

**Comment by lukaslueg** [comment]
Yes. This had always been unstable, but now `rustc` seems to finally re-order fields.
